### PR TITLE
a compiling chat protocol

### PR DESCRIPTION
**Pull request by maxtaco**
- basic local and remote protocols
- some guessees as to how the crypto will work
- pull over SignatureInfo and Ciphertext format from KBFS

### CODE REVIEW

**Comment by maxtaco** [comment]
@mmaxim do you want to take another look at this?  New changes incorporate that people want things signed.

Also, @akalin-keybase I want to try to match the KBFS signature/encryption scheme as closely as possible.

File: protocol/avdl/chat_common.avdl [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  // The same format as in KBFS (see libkbfs/data_types.go)
+  record EncryptedData {
+    int   v;  // version = 1
+    bytes e;  // encryptedData (output of secret box)
+    bytes n;  // nonce
+  }
+
+  record SignatureInfo {
+    int   v; // version = 1
+    bytes s; // signature; output if EdDSA
```
Comment (akalin-keybase) [inline protocol/avdl/chat_common.avdl]: if -> of

**Comment by akalin-keybase** [comment]
KBFS-matching stuff looks good to me.

**Comment by mmaxim** [comment]
LGTM!

**Comment by maxtaco** [comment]
This turned into a gregor PR, see over there...

**Pull request closed without merging**